eng/pipelines/sdk-update.yml

```diff
@@ -13,7 +13,6 @@ resources:
       type: github
       name: Azure/azure-sdk-tools
       endpoint: azure
-      ref: refs/tags/azure-sdk-tools_20220404.3
```

Comment:
Why was this removed? Looks like it was added back with https://github.com/Azure/autorest.csharp/commit/768a83a695500265d208e757d5ea0a6443ebfac2 which is good but we do want to keep these refs instead of floating.